### ISSUE TITLE
Spack --dirty in configuration

### ISSUE THREAD

**Issue by citibeth** [LLNL/spack#1925]
There needs to be a way to configure Spack so when you say `spack install`, `spack diy`, etc. --- it does it in "dirty" mode, but without you having to write `spack install --dirty` on every command line.

The configuration should probably be taken out of one of the files in the `.spack` directory.

**Comment by adamjstewart** [comment]
Users should be able to set default flags for any command. I would personally love to tell `spack install` to always be verbose.

**Comment by citibeth** [comment]
You are absolutely right.  Any proposals on the UI for this?  I'll throw
one out... suppose I create `~/.spack/defaults.yaml` like:
```
defaults:
    install: [-v --dirty]
    diy: [--dirty]
    setup: [--dirty]
```

I suppose this might get a bit verbose because certain flags are used again
and again for multiple commands.  But I still like it... simple,
straightforward, flexible.

Any sketches on how to go about doing this?

**Comment by tgamblin** [comment]
@adamjstewart: Just out of curiosity did you know you can hit `v` when you type `spack install` without `-v` and it'll become verbose?

**Comment by adamjstewart** [comment]
@tgamblin Yeah, I think I discovered that by accident forever ago, or you told it to me, I don't remember. But now I just compulsively add the `-v`

**Comment by adamjstewart** [comment]
@citibeth I wouldn't call it `defaults.yaml`, since `packages.yaml` is basically defaults as well. I would add it to a generic `configuration.yaml` with all of the other things that we've been talking about adding global configurations for.

**Comment by tgamblin** [comment]
@citibeth @adamjstewart: do you know of *any* tool that has this feature? I suppose `ls` and `grep` have `*_OPTIONS` env vars.  I'd put it in the nice-to-have category but there are a lot of other things that need adding...

**Comment by tgamblin** [comment]
Clarification: I think it's fine to have this for `--dirty`.  I am not so sure that it's worth implementing generally for everything.

**Comment by citibeth** [comment]
Git has this feature, you can set things in `~/.gitconfig`.  For example, see:

https://github.com/citibeth/spack/commit/fabeaa8f6f3ebd6473b21e40ebd47be9c921c688#diff-3987f26e56771d3a4a844a6990abc25a

**Comment by tgamblin** [comment]
Git has this implemented for particular features.  The part that scares me is "all commands".

I don't think all features map neatly to command options -- e.g. for this one do you really want to have to write 3 lines in your config file to enable dirty builds?  Why not a single option for dirty installs in the config?

Same for verbose -- why not a single default verbose flag?

**Comment by adamjstewart** [comment]
I agree that it isn't very crucial. I'd rather fix bugs than add new features. I don't think the implementation matters too much. I like the way `.gitconfig` works.

One thing that will be problematic is that the option flags might change, meaning Spack won't be able to read your configuration file any more until it is updated.

**Comment by tgamblin** [comment]
Fixed by #2152
